taccsite_cms/contrib/taccsite_sysmon/static/taccsite_sysmon/js/sysmon.js

```
@@ -0,0 +1,168 @@
+
+/**
+ * The URL of the API endpoint
+ * @type {string}
+ */
+const API_URL = '/api/system-monitor';
+
+/**
+ * The systems to show
+ *
+ * _Notice: This value is expected to be available from another script_
+ * @type {string}
+ */
+const SYSTEM_HOSTNAME = window.SYSMON_SYSTEM_HOSTNAME;
+
+/**
+ * Load system status
+ * @param {string} path
+ * @param {function} onSuccess - Callback for success (receives JSON)
+ * @param {function} onError - Callback for success (receives XMLHttpRequest)
+ */
+function loadStatus(path, onSuccess, onError) {
+  var xhr = new XMLHttpRequest();
+  xhr.onreadystatechange = function () {
+    if (xhr.readyState === XMLHttpRequest.DONE) {
+      if (xhr.status === 200) {
+        if (onSuccess) onSuccess(JSON.parse(xhr.responseText));
+      } else {
+        if (onError) onError(xhr);
+      }
+    }
+  };
+  xhr.open('GET', path, true);
+  xhr.send();
+}
+
+/**
+ * Whether system is operational
+ * @param {System} system
+ * @return {boolean}
+ */
+function isOperational(system) {
+  if (system['load_percentage'] < 1 || system['load_percentage'] > 99) {
+    system['load_percentage'] = 0;
+    return system['jobs']['running'] > 1;
+  }
+  return true;
+}
+
+/**
+ * Show system content in UI
+ */
+function showStatus() {
+  document.getElementById('status').classList.remove('d-none');
+}
+
+/**
+ * Style system status
+ * @param {string} type - A type: "warning"
+ */
+function setStatusStyle(type) {
+  switch (type) {
+    case 'warning':
```

Will there be more `type`s to compare against? If not, is there reason that `switch` used here (in comparison to other blocks?

---

I began the work for this, but then abandoned it, becuase it grew beyond the scope of a "minimal" plugin. I have added a requirement on #295 to support multiple, distinct states (rather than assume failure and ignore loading).